src/pages/studyView/table/GenePanelModal.tsx

```diff
@@ -40,25 +45,41 @@ export const GenePanelList: React.FunctionComponent<IGenePanelTooltipProps> = ({
     }
 };
 
-export const GenePanelModal: React.FunctionComponent<IGeneModalProps> = ({
-    show,
-    panelName,
-    genes,
-    hide
-}) => {
-    return (
-        <Modal show={show} onHide={hide} keyboard>
-            <Modal.Header closeButton>
-                <Modal.Title>{panelName}</Modal.Title>
-            </Modal.Header>
-            <Modal.Body style={{ maxHeight: "calc(100vh - 210px)", overflowY: "auto" }}>
-                {genes.map(gene => (
-                    <p key={gene.entrezGeneId}>{gene.hugoGeneSymbol}</p>
-                ))}
-            </Modal.Body>
-            <Modal.Footer>
-                <Button onClick={hide}>Close</Button>
-            </Modal.Footer>
-        </Modal>
-    );
+@observer
+export class GenePanelModal extends React.Component<IGeneModalProps, {}> {
+    @observable dataLoading = false;
+    @observable genes: GenePanelToGene[] = [];
+
+    @autobind
+    componentWillReceiveProps(nextProps: Readonly<IGeneModalProps>): void {
```

Comment:
why do you want to do it like this instead of using a mobxpromise? seems more complicated this way

Reply:
one construct that might help which you may not be aware of is MobxPromiseCache, which handles the case here - you would pass in the panel name, and it would create a promise based on that panel name. Let me know if you want to talk about it